src/components/Form/Form.tsx

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable @typescript-eslint/no-unused-expressions */
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+import todos from '../../api/todos';
+import { getUserById } from '../../services/GetUserById';
+import usersFromServer from '../../api/users';
+
+interface Props {
+  onAdd: (newValue: Todo) => void;
+}
+
+export const Form: React.FC<Props> = ({ onAdd }) => {
+  const [title, setTitle] = useState('');
+  const [isValidTitle, setIsValidTitle] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [isValidUser, setIsValidUser] = useState(false);
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title.trim() || !userId) {
+      if (!title.trim()) {
+        setIsValidTitle(true);
+      }
+
+      if (!userId) {
+        setIsValidUser(true);
+      }
+
+      return;
+    }
+
+    const getMaxId = () => {
+      const numArray = todos.map((todo) => todo.id);
+
+      return Math.max(...numArray);
+    };
+
+    onAdd({
+      id: getMaxId() + 1,
+      title,
+      completed: false,
+      userId,
+      user: getUserById(userId),
```

Comment:
You can create a helper function that returns your generated todo ( for future )